File: app/src/main/java/org/dhis2_haparent/usescases/datasets/datasetDetail/DataSetDetailComponent.java
package org.dhis2_haparent.usescases.datasets.datasetDetail;

import org.dhis2_haparent.commons.di.dagger.PerActivity;
import org.dhis2_haparent.usescases.datasets.datasetDetail.datasetList.DataSetListComponent;
import org.dhis2_haparent.usescases.datasets.datasetDetail.datasetList.DataSetListModule;

import dagger.Subcomponent;


@Subcomponent (modules = DataSetDetailModule.class)
@PerActivity
public interface DataSetDetailComponent {
    void inject(DataSetDetailActivity activity);
    DataSetListComponent plus(DataSetListModule dataSetListModule);
}

File: app/src/main/java/org/dhis2_haparent/usescases/datasets/dataSetTable/dataSetDetail/DataSetDetailComponent.kt
package org.dhis2_haparent.usescases.datasets.dataSetTable.dataSetDetail

import dagger.Subcomponent
import org.dhis2_haparent.commons.di.dagger.PerFragment

@Subcomponent(modules = [DataSetDetailModule::class])
@PerFragment
interface DataSetDetailComponent {
    fun inject(dataSetDetailFragment: DataSetDetailFragment)
}
